doc/src/manual/methods.md

```diff
@@ -515,6 +515,226 @@ julia> wait(schedule(t, 1))
 "definition for Int"
 ```
 
+## Design Patterns with Parametric Methods
+
+
+While complex dispatch logic is not required for performance or usability,
+sometimes it can be the best way to express some algorithm.
+Here are a few common design patterns that come up sometimes when using dispatch in this way.
+
+### Extracting the type parameter from a super-type
+
+
+Here is the correct code template for returning the element-type `T`
+of any arbitrary subtype of `AbstractArray`:
+
+```julia
+abstract AbstractArray{T, N}
+eltype(::Type{<:AbstractArray{T}}) where {T} = T
+```
+using so-called triangular dispatch.
+
+Another way, which used to be the only correct way before the advent of
+triangular dispatch, is:
+
+```julia
+abstract AbstractArray{T, N}
+eltype(::Type{AbstractArray}) = Any
+eltype{T}(::Type{AbstractArray{T}}) = T
+eltype{T, N}(::Type{AbstractArray{T, N}}) = T
+eltype{A<:AbstractArray}(::Type{A}) = eltype(supertype(A))
+```
```

Comment:
Yes. I just wanted to show both in the manual so that people could make an informed choice :)

Also, I did indeed miss a class of cases:
```julia
julia> f(Array{Array{T,1}, 1} where T)
ERROR: UndefVarError: T not defined
Stacktrace:
 [1] f(::Type{Array{Array{T,1},1} where T}) at ./REPL[1]:1
```
This is the case that shows why it needs `@isdefined(T) ? T : Any`.

(this also triggers what appears to be a type-system bug in "my" old version for which I'll file a separate issue)